### PR TITLE
Use preinstalled msys2

### LINKED ISSUE #65767: Remove msys2 hacks and vendor the whole thing
On 2019-10-24 we had a CI outage due to the ca-certificates msys2 package being broken. The hack @mati865 implemented (:heart:) is to download an older, known good package beforehand and install it after msys2 updated its packages.

Once https://github.com/msys2/MINGW-packages/issues/5887 is fixed we should revert the hack, but I think at this point it's worth investing in vendoring msys2 as a whole.

### PR DESCRIPTION
**Pull request by mati865**
Fixes https://github.com/rust-lang/rust/issues/65767

### CODE REVIEW

**Comment by mati865** [comment]
r? @pietroalbini 

Successful CI builds: https://dev.azure.com/rust-lang/rust/_build/results?buildId=31583&view=logs&j=b1e7928f-7d26-57cb-a49e-81353e6c69ce and https://github.com/rust-lang/rust/pull/73188/checks?check_run_id=755856941

File: src/ci/scripts/install-msys2-packages.sh [hunk 1]
```diff
@@ -6,9 +6,6 @@ IFS=$'\n\t'
 source "$(cd "$(dirname "$0")" && pwd)/../shared.sh"
 
 if isWindows; then
-    pacman -S --noconfirm --needed base-devel ca-certificates make diffutils tar \
-        binutils
-
     # Detect the native Python version installed on the agent. On GitHub
```
Comment (pietroalbini) [inline src/ci/scripts/install-msys2-packages.sh:9]: Can you fold this whole file in the `install-msys2.sh` script? The only reason this was executed in a separate script was incompatibilities with git bash and msys2's bash, and that's not a problem anymore.

**Review by emilyalbini — CHANGES_REQUESTED** [review]

**Comment by mati865** [comment]
BTW there is bug when running `./x.py run src/tools/expand-yaml-anchors` on Windows:
```
diff --git a/.github/workflows/ci.yml b/.github/workflows/ci.yml
index 355f2829215..2d62df39459 100644
--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -5,7 +5,7 @@
 # This file was automatically generated by the expand-yaml-anchors tool. The
 # source file that generated this one is:
 #
-#   src/ci/github-actions/ci.yml
+#   src/ci/github-actions\ci.yml
 #
 # Once you make changes to that file you need to run:
 #
```

**Comment by pietroalbini** [comment]
This looks good now! Thanks!

@bors r+ rollup=never

> BTW there is bug when running ./x.py run src/tools/expand-yaml-anchors on Windows:

Oh that's bad, do you have the time to open a PR fixing that?

**Comment by mati865** [comment]
> Oh that's bad, do you have the time to open a PR fixing that?

I'll look into `expand-yaml-anchors` code and open PR or issue depending on how long it takes to fix it 😄

**Comment by pietroalbini** [comment]
> I'll look into expand-yaml-anchors code and open PR or issue depending on how long it takes to fix

I'm happy to mentor: if you need any help feel free to ping me wherever.

**Comment by mati865** [comment]
Damn, I think I saw open issue about it but there are dozens of them when searching for `path label:O-windows`:
```
use std::path::Path;

fn main() {
    let path = Path::new("/tmp/foo").join("bar.rs");
    println!("{}", path.display());
}
```
This gives:
```
$ rustc test.rs && ./test.exe
/tmp/foo\bar.rs
```

**Pull request merged**